src/parser/tests/class_parser_tests.rs

```diff
@@ -0,0 +1,225 @@
+    let method = &unit.implementations[0];
+    assert_eq!(unit.implementations.len(), 1);
+
+    assert_eq!(method_pou.name, "testMethod3");
+    assert_eq!(method.access, Some(AccessModifier::Private));
+    assert_eq!(method_pou.poly_mode, Some(PolymorphismMode::Abstract));
+    assert_ne!(method_pou.return_type, None);
+    assert_eq!(method.overriding, true);
+}
+
+#[test]
+fn class_with_var_default_block() {
+    let lexer = lex("CLASS MyClass VAR END_VAR END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 0);
+
+    let vblock = &class.variable_blocks[0];
+    assert_eq!(vblock.variables.len(), 0);
+
+    assert_eq!(vblock.retain, false);
+    assert_eq!(vblock.constant, false);
+    assert_eq!(vblock.access, AccessModifier::Protected);
+    assert_eq!(vblock.variable_block_type, VariableBlockType::Local);
+}
+
+#[test]
+fn class_with_var_non_retain_block() {
+    let lexer = lex("CLASS MyClass VAR CONSTANT NON_RETAIN PUBLIC END_VAR END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 0);
+
+    let vblock = &class.variable_blocks[0];
+    assert_eq!(vblock.variables.len(), 0);
+
+    assert_eq!(vblock.retain, false);
+    assert_eq!(vblock.constant, true);
+    assert_eq!(vblock.access, AccessModifier::Public);
+    assert_eq!(vblock.variable_block_type, VariableBlockType::Local);
+}
+
+#[test]
+fn class_with_var_retain_block() {
+    let lexer = lex("CLASS MyClass VAR RETAIN INTERNAL END_VAR END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 0);
+
+    let vblock = &class.variable_blocks[0];
+    assert_eq!(vblock.variables.len(), 0);
+
+    assert_eq!(vblock.retain, true);
+    assert_eq!(vblock.constant, false);
+    assert_eq!(vblock.access, AccessModifier::Internal);
+    assert_eq!(vblock.variable_block_type, VariableBlockType::Local);
+}
+
+#[test]
+fn method_with_var_block() {
+    let lexer = lex("CLASS MyClass METHOD testMethod3 VAR END_VAR END_METHOD END_CLASS");
+    let unit = parse(lexer).0;
+
+    let class = &unit.units[0];
+    assert_eq!(class.pou_type, PouType::Class);
+    assert_eq!(unit.implementations.len(), 1);
+
+    let method_pou = &unit.units[1];
+    let vblock = &method_pou.variable_blocks[0];
+
+    assert_eq!(vblock.retain, false);
+    assert_eq!(vblock.constant, false);
+    assert_eq!(vblock.access, AccessModifier::Protected);
+    assert_eq!(vblock.variable_block_type, VariableBlockType::Local);
+}
+
+#[test]
+fn method_with_var_inout_blocks() {
+    let lexer = lex(
+        "CLASS MyClass METHOD testMethod3 VAR_INPUT CONSTANT x:SINT := 3; END_VAR VAR_IN_OUT END_VAR VAR_OUTPUT END_VAR END_METHOD END_CLASS",
```

Comment:
Could you please split this in multiple lines?

Reply:
Done!